cmd/kubelet/app/server.go

```diff
@@ -595,7 +596,7 @@ func run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 		if closeAllConns == nil {
 			return errors.New("closeAllConns must be a valid function other than nil")
 		}
-		kubeDeps.OnHeartbeatFailure = closeAllConns
+		kubeDeps.OnHeartbeatFailure = createHeartbeatFailureHandler(closeAllConns, 5 * time.Minute)
```

Comment:
is this still needed after https://github.com/kubernetes/kubernetes/pull/104844? that PR (in 1.23) switched to default to only closing idle connections, trusting the http/2 ping to close stuck open connections

Reply:
Good question.  Checking my notes I see that I did one test with that change in place, on a single node out of the 5000, observed that the problem still happened on that node, and concluded that PR didn't help.  But, now that I understand the cause better, I see that I should have applied it to all nodes to improve the overall load pattern. (As I did later with the fix in this PR).  I will have to re-test with 104844.  Thanks @liggitt

Reply:
See main comment thread for retest result and discusion. Short answer is yes, it is still needed :-)

Reply:
I guess I don't fully understand why that PR doesn't fix the problem (i.e. what exactly --rotate-certificates=true changes wrt to that). Can someone clarify that further? @aojea - FYI

[But assuming that indeed it's not addressed, this change makes sense to me.]